Tasks/Maven/mavencodeanalysis.ts

```diff
@@ -0,0 +1,231 @@
+/// <reference path="../../definitions/vsts-task-lib.d.ts" />
+/// <reference path="../../definitions/Q.d.ts" />
+
+import path = require('path');
+import fs = require('fs');
+
+import tl = require('vsts-task-lib/task');
+import trm = require('vsts-task-lib/toolrunner');
+
+// Lowercased names are to lessen the likelihood of xplat issues
+import pmd = require('./pmdformaven');
+import ar = require('./analysisresult');
+import ma = require('./moduleanalysis');
+
+// Set up for localization
+tl.setResourcePath(path.join( __dirname, 'task.json'));
+
+// Cache build variables - if they are null, we are in a test env and can use test inputs
+// The artifact staging directory will be a subdirectory just to be safe.
+var sourcesDir:string = tl.getVariable('build.sourcesDirectory') || tl.getInput('test.sourcesDirectory');
+var stagingDir:string = path.join(tl.getVariable('build.artifactStagingDirectory') || tl.getInput('test.artifactStagingDirectory'), ".codeAnalysis");
+var buildNumber:string = tl.getVariable('build.buildNumber') || tl.getInput('test.buildNumber');
+
+// Apply goals for enabled code analysis tools
+export function applyEnabledCodeAnalysisGoals(mvnRun: trm.ToolRunner):void {
+    var enabledCodeAnalysisTools:string[] = getEnabledCodeAnalysisTools();
```

Comment:
Instead of getting the tools and checking a condition just refactor to isCodeAnalysisToolEnabled(toolName) and use it here.